### ISSUE TITLE
Central transaction table for all payment gateways

### ISSUE THREAD

**Issue by miha1** [PrestaShop/PrestaShop#13861]
#### Is your feature request related to a problem?
As part of the new [3-D Secure](https://doc.wirecard.com/CreditCard.html#CreditCard_3DS2) standard that is coming into force in Europe, there are a number of new parameters that should be sent to credit card acquirers.

Among them are parameters that relate to the customers transactions within a given time period (e.g. in the previous day or within the previous year). As there is no central transaction table in Prestashop that is currently tracking this, as all payment plugins use different standards.

#### Describe the solution you'd like
Ideally there would be a new database table and backend view in which all payment transactions that happened within the Prestashop (successful and failed transaction) can be listed and where payment providers can add any transactions that they fulfill.

It should provide information about what user has done with transaction at which point in time.

**Comment by khouloudbelguith** [comment]
Hi @miha1,

From the BO, you need to display the list of the customers' transactions within a given time period?
Thanks!

**Comment by miha1** [comment]
Yes, I need the statistic how much transaction are tried, successful, void with dates

**Comment by khouloudbelguith** [comment]
@miha1, in the BO => Orders => Orders page, you can filter by `Date` & by `customer`
![image](https://user-images.githubusercontent.com/16067358/58010644-aaea6480-7ae8-11e9-9766-8cf47f12ab32.png)
Thanks to check & feedback

**Comment by miha1** [comment]
Thanks for the screenshot but just to clarify. I only see one transaction for one order right? The new standards targets to see all transactions made for one order. Did it fail, was it partly done by card and partly by voucher. So one order can have several transactions. Will Prestashop support this in future?

**Comment by khouloudbelguith** [comment]
@miha1, so you need to get the number of transactions for each order in a limited period?
Thanks!

**Comment by miha1** [comment]
Not only number, but all information about tried transaction for each order

**Comment by khouloudbelguith** [comment]
@miha1, thanks for your feedback.
ping @marionf, @colinegin what do you think of his suggestion?
Thanks!

**Comment by Hlavtox** [comment]
My 5 cents - I think that we could extend the ps_order_payments table with a column for extra data. That would suffice.
